filesystem/iso9660.ksy

```diff
@@ -216,60 +94,867 @@ types:
-        encoding: ASCII
-      - id: month
-        type: str
-        size: 2
-        encoding: ASCII
-      - id: day
-        type: str
-        size: 2
-        encoding: ASCII
-      - id: hour
-        type: str
-        size: 2
-        encoding: ASCII
-      - id: minute
-        type: str
-        size: 2
-        encoding: ASCII
-      - id: sec
-        type: str
-        size: 2
-        encoding: ASCII
-      - id: sec_hundreds
-        type: str
-        size: 2
-        encoding: ASCII
-      - id: timezone
+      - id: type
+        -orig-id: volume_descriptor_type
+        doc-ref: ecma-119 8.1.1
         type: u1
-  u2bi:
-    seq:
-      - id: le
-        type: u2le
-      - id: be
-        type: u2be
-  u4bi:
-    seq:
-      - id: le
-        type: u4le
-      - id: be
-        type: u4be
-instances:
-  sector_size:
-    value: 2048
-  primary_vol_desc:
-    pos: 0x010 * sector_size
-    type: vol_desc
+        enum: volume_type
+      - id: magic
+        doc-ref: ecma-119 8.1.2
+        contents: [ 0x43, 0x44, 0x30, 0x30, 0x31 ]
+      - id: version
+        doc-ref: ecma-119 8.1.3
+        type: u1
+        valid: 1
+      - id: volume
+        type:
+          switch-on: type
+          cases:
+            'volume_type::boot_record_volume_descriptor': boot_record
+            'volume_type::primary_volume_descriptor': primary
+            'volume_type::supplementary_volume_descriptor': supplementary
+            #'volume_type::volume_partition_descriptor': partition
+    types:
+      text32: # 0x20
+        seq:
+          - id: text
+            type: str
+            size: 0x20
+      text37: # 0x25
+        seq:
+          - id: text
+            type: str
+            size: 0x25
+      text128: # 0x80
+        seq:
+          - id: text
+            type: str
+            size: 0x80
+      boot_record:
+        doc-ref: ecma-119 8.2
+        seq:
+          - id: boot_system_identifier
+            doc-ref: ecma-119 8.2.4
+            type: strz
+            size: 0x20
+          - id: boot_identifier
+            doc-ref: ecma-119 8.2.5
+            type: strz
+            size: 0x20
+      primary:
+        doc-ref: ecma-119 8.4
+        seq:
+          - id: unused01
+            doc-ref: ecma-119 8.4.4
+            type: u1
+            valid: 0
+          - id: system_identifier
+            doc-ref: ecma-119 8.4.5
+            type: text32
+          - id: volume_identifier
+            doc-ref: ecma-119 8.4.6
+            type: text32
+          - id: unused02
+            doc-ref: ecma-119 8.4.7
+            contents: [ 0x0, 0x0, 0x0, 0x0,  0x0, 0x0, 0x0, 0x0 ]
+          - id: volume_space_size
+            doc-ref: ecma-119 8.4.8
+            type: u4bi
+          - id: unused03
+            doc-ref: ecma-119 8.4.9
+            contents: [ 0x0, 0x0, 0x0, 0x0,  0x0, 0x0, 0x0, 0x0,  0x0, 0x0, 0x0, 0x0,  0x0, 0x0, 0x0, 0x0,  0x0, 0x0, 0x0, 0x0,  0x0, 0x0, 0x0, 0x0,  0x0, 0x0, 0x0, 0x0,  0x0, 0x0, 0x0, 0x0 ]
```

Comment:
Please split it into multiple lines for better readability:

```suggestion
            contents: |
              [
                0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
                0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
                0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
                0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0
              ]
```

Reply:
IMHO `[0x00] * 32` would be much more readable

Reply:
fixed

Reply:
I had to revert this!
It breaks while testing on https://ide.kaitai.io/

Reply:
> I had to revert this!
> It breaks while testing on https://ide.kaitai.io/

Oh, you're right, thanks for pointing it out! That's actually something predictable, since `contents` only supports YAML arrays, not string expressions as most KS language keys.

In this case, just leave out the `|` character which makes the contents to be interpreted as a string:

```suggestion
            contents: [
              0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
              0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
              0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
              0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0
            ]
```

Reply:
Fixed, but... the closing bracket on its own line also broke it.